dulwich/objectspec.py

```diff
@@ -54,6 +54,8 @@ def parse_tree(repo, treeish):
       KeyError: If the object can not be found
     """
     treeish = to_bytes(treeish)
+    if treeish in repo.refs.keys(base=b"refs/heads/"):
```

Comment:
perhaps just call parse_ref() here, so it also e.g. handles tags?

Reply:
using parse_ref() now